corehq/apps/hqwebapp/apps.py

```diff
@@ -7,3 +8,35 @@ class HqWebAppConfig(AppConfig):
     def ready(self):
         # Ensure the login signal handlers have been loaded
         from . import login_handlers, signals  # noqa
+        # custom 2FA methods need to be registered on startup
+        register_two_factor_methods()
+
+
+def register_two_factor_methods():
+    from two_factor.plugins.registry import registry
+
+    from corehq.apps.hqwebapp.two_factor_methods import (
+        HQGeneratorMethod,
+        HQPhoneCallMethod,
+        HQSMSMethod,
+    )
+
+    # default generator method is registered when the registry object is created
+    # https://github.com/jazzband/django-two-factor-auth/blob/1.15.5/two_factor/plugins/registry.py#L76-L77
+    registry.unregister('generator')
+    registry.register(HQGeneratorMethod())
+
+    # default phone methods are registered when django starts up
+    # https://github.com/jazzband/django-two-factor-auth/blob/master/two_factor/plugins/phonenumber/apps.py#L19-L30
+    if not settings.ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE:
+        registry.unregister('call')
+        registry.unregister('sms')
+        return
+
+    if getattr(settings, 'TWO_FACTOR_CALL_GATEWAY', None):
+        registry.unregister('call')
```

Comment:
Just confirming that calling `unregister` on a method that wasn't registered originally won't be an issue?

Reply:
Yeah here's the [implementation](https://github.com/jazzband/django-two-factor-auth/blob/1.15.5/two_factor/plugins/registry.py#L86-L87) in django-two-factor-auth for v1.15.5. It just filters the list of methods to exlude any matching the key passed in, in this case 'call'.